src/edu/washington/escience/myria/io/UriSource.java

```diff
@@ -51,14 +52,24 @@ public UriSource(@JsonProperty(value = "uri", required = true) final String uri)
 
   @Override
   public InputStream getInputStream() throws IOException {
-    // Use Hadoop's URI parsing machinery to extract an input stream for the underlying URI
+    URI parsedUri = URI.create(uri);
+
+    return parsedUri.getScheme().equals("http") || parsedUri.getScheme().equals("https")
```

Comment:
I tend to avoid trigraphs unless they're absolutely necessary, but that's up to you.

However, please `()` around the condition of `||` here... the code is correct but you have to know non-obvious parts of [the precedence chart (13 vs 14)](http://introcs.cs.princeton.edu/java/11precedence/).

LGTM!